core/src/main/java/jenkins/util/xml/XMLUtils.java

```diff
@@ -29,6 +43,31 @@
     private final static Logger LOGGER = LogManager.getLogManager().getLogger(XMLUtils.class.getName());
     private final static String DISABLED_PROPERTY_NAME = XMLUtils.class.getName() + ".disableXXEPrevention";
 
+    public static final String FEATURE_HTTP_XML_ORG_SAX_FEATURES_EXTERNAL_GENERAL_ENTITIES = "http://xml.org/sax/features/external-general-entities";
+    public static final String FEATURE_HTTP_XML_ORG_SAX_FEATURES_EXTERNAL_PARAMETER_ENTITIES = "http://xml.org/sax/features/external-parameter-entities";
+
+    private static final DocumentBuilderFactory documentBuilderFactory;
```

Comment:
A DocumentBuilderFactory is not gauranteed to be thread safe - so keeping this as a field is not wise.